File: 2017/1/one.py
i_n = open('input.txt', 'r').read()

def part_one(input):
    s = 0
    for i in range(len(i_n)):
        if i_n[i] == i_n[(i + 1) % len(i_n)]:
            s += int(i_n[i])
    return s

def part_two(input):
    s = 0
    for i in range(len(i_n)):
        if i_n[i] == i_n[(i + (len(i_n)//2)) % len(i_n)]:
            s += int(i_n[i])
    return s


print(f'pt 1: {part_one(i_n)}, pt 2: {part_two(i_n)}.')

File: 2024/1/one.lua
local file = io.open("input.txt", "r")

local list1 = {}
local list2 = {}

for line in file:lines() do
    local part1, part2 = line:match("(%S+)%s+(%S+)")
    table.insert(list1, part1)
    table.insert(list2, part2)
end

file:close()

table.sort(list1)
table.sort(list2)
local s = 0 

for i,v in ipairs(list1) do
  s = s + math.abs(v - list2[i])
end

print(s)

